pkg/collector/corechecks/net/networkv2/network.go

```diff
@@ -442,11 +450,44 @@ func getEthtoolMetrics(driverName string, statsMap map[string]uint64) map[string
 				queueTag = "queue:" + queueNum
 				newKey = metricName
 				metricPrefix = ".queue."
+				continueCase = false
 			} else {
 				continueCase = true
 			}
 		}
 		if continueCase {
+			if strings.HasPrefix(statName, "rx") || strings.HasPrefix(statName, "tx") {
+				// Extract the queue and the metric name from ethtool stat name:
+				//   tx0_bytes -> (queue:0, tx_bytes)
+				//   rx1_packets -> (queue:1, rx_packets)
+				parts := strings.Split(statName, "_")
+				queueIndex := -1
+				queueNum := -1
+				for i, part := range parts {
+					if !strings.HasPrefix(part, "rx") && !strings.HasPrefix(part, "tx") {
+						continue
+					}
+					if cur, err := strconv.Atoi(part[2:]); err == nil {
+						queueIndex = i
+						queueNum = cur
+						break
+					}
+				}
+				if queueIndex == -1 {
+					continueCase = true
+				} else {
+					parts = append(append([]string{}, parts[queueIndex][:2]), parts[queueIndex+1:]...)
```

Comment:
I don't understand the need for the extra `append` here (I'm not a golang expert so this could be a me issue...). Why is this preferable to 
```
parts = append([]string{parts[queueIndex][:2]}, parts[queueIndex+1:]...)
```
?

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/40326/commits/f9301b7a8ddafbc920c7a5ed79a33d8368330b55